src/lib/network/network_autoyast.rb

```diff
@@ -206,16 +222,53 @@ def merge_routing(instsys_routing, ay_routing)
       instsys_routing.merge(ay_routing)
     end
 
-    # Returns networking section of current AY profile
-    def ay_networking_section
+    # Returns current AY profile in the internal representation
+    #
+    # @return [Hash] hash representing current profile or empty hash
+    def ay_current_profile
       Yast.import "Profile"
 
       ay_profile = Profile.current
 
       return {} if ay_profile.nil? || ay_profile.empty?
-      return {} if ay_profile["networking"].nil?
+      ay_profile
+    end
+
+    # Returns networking section of current AY profile
+    def ay_networking_section
```

Comment:
in my POV "networking" section is used quite often so it deserved own method as a shortcut.